### PR TITLE
Add reasons why we don't share feedback

### PR DESCRIPTION
**Pull request by nicksnyder**
This refines and expands the content added in https://github.com/sourcegraph/about/pull/445.

### CODE REVIEW

File: handbook/people-ops/hiring.md [hunk 1]
```diff
@@ -16,16 +16,31 @@ We use [Google Hire](https://hire.withgoogle.com/t/sourcegraphcom/hiring/landing
...
+
+- Applications
+    - Does this candidate meet all qualifications for the role they are applying for?
+    - Does this candidate's resume and application effectively communicate their skills, accomplishments, and a genuine interest in Sourcegraph?
+    - How does this candidate compare to other candidate's in our pipeline and our current team?
+- Interviews
+    - Does this candidate have the skills and experience that we are looking for?
+    - How does this candidate compare to other candidate's in our pipeline and our current team?
+    - What makes us excited about hiring this candidate?
+    - Is this candidate in the top N candidates for the role (where N is the number of hires we want to make)?
```
[the remark this answers is not in the document]
Comment (nicksnyder) [reply handbook/people-ops/hiring.md]: I will remove this. It is kind of redundant with the second bullet point.
```suggestion
```

File: handbook/people-ops/hiring.md [hunk 2]
```diff
@@ -16,16 +16,31 @@ We use [Google Hire](https://hire.withgoogle.com/t/sourcegraphcom/hiring/landing
...
 
-## Sharing feedback
+## Decision making process
+
+We review each candidate at each stage of our hiring pipeline (e.g. application, phone screen, interviews, references) to determine if we want to move the candidate to the next stage. Here some things that we consider:
+
+- Applications
+    - Does this candidate meet all qualifications for the role they are applying for?
+    - Does this candidate's resume and application effectively communicate their skills, accomplishments, and a genuine interest in Sourcegraph?
+    - How does this candidate compare to other candidate's in our pipeline and our current team?
```
[the remark this answers is not in the document]
Comment (nicksnyder) [reply handbook/people-ops/hiring.md]: The point I am trying to make is that we don't interview candidates in a vacuum. If we have capacity to interview and hire N people for a role in a given timeframe and there are M candidates in our pipeline (where M >> N) then we are going to prioritize interviewing the best candidates at each stage of the interview process. I will think about a better way to phrase this, but let me know if you have any concrete suggestions.

[the remark this answers is not in the document]
Comment (mercadon) [reply handbook/people-ops/hiring.md]: I'd like to think of it as something along the lines of what the candidate is adding to the current team. Perhaps something along the lines of "what skills and experience does this candidate bring to the table that our current team does not currently have?"

[the remark this answers is not in the document]
Comment (nicksnyder) [reply handbook/people-ops/hiring.md]: Updated based on Dan's suggestion. 

"what skills and experience does this candidate bring to the table that our current team does not currently have?" seemed a bit redundant with "Does this candidate have the skills and experience that we are looking for?".

**Pull request merged**